### PR TITLE
Mastermind Javascript game added - GSSOC'23

### LINKED ISSUE #2414: Mastermind game
**Describe the project you want to add with tech stack**
The Mastermind game is a classic memory-based game implemented using HTML, CSS, and JavaScript. The objective of the game is to flip over pairs of cards and find matching pairs. The game continues until all cards have been matched.

**Expected behavior**
Following are the features of the game:
Interactive grid of cards: The game presents a grid of face-down cards that can be flipped by clicking on them.
Flipping cards: Players can click on two cards to flip them over and reveal their contents.
Match tracking: The game keeps track of flipped cards and checks for matches. If a match is found, the cards stay flipped; otherwise, they are turned back face-down.
Game completion: The game ends when all cards have been matched.
Timer: A timer displays the duration of the gameplay, allowing players to track their performance.
Move counter: The game tracks the number of moves made by the player, providing a measure of progress.

Please assign it to me as part of GSSOC'23

### PR DESCRIPTION
**Pull request by srujana-16**
# Mastermind game - new project 

Close #2414 

# Checklist:

- [x] I have mentioned the issue number in my Pull Request.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`

### CODE REVIEW

**Comment by srujana-16** [comment]
Please merge this as part of GSSOC'23

**Comment by srujana-16** [comment]
Hello! Please check this PR and merge it as part of GSSOC'23

File: Js-Games/Mastemind/Screenshots/Mastermind.png
Comment (MohitGupta121) [inline Js-Games/Mastemind/Screenshots/Mastermind.png:1]: @srujana-16 please remove the image from PR and upload it somewhere in cloud and use the link of the image to load the image into your Project.

**Review by MohitGupta121 — CHANGES_REQUESTED** [review]
@srujana-16 also resolve conflicts

**Comment by srujana-16** [comment]
I have made the changes and resolved the conflicts, please check

**Review by MohitGupta121 — APPROVED** [review]

**Review by MohitGupta121 — CHANGES_REQUESTED** [review]
@srujana-16 still there are conflicts

**Comment by srujana-16** [comment]
Please check now

**Review by MohitGupta121 — APPROVED** [review]

**Pull request merged**